### PR TITLE
fix: filter values of depends on property select

### PR DESCRIPTION
**Pull request by clagosarias**
### [ADF-797](https://oat-sa.atlassian.net/browse/ADF-797)

**Description**
Prevent user from being able to create multiple level dependencias.

I have achieved this by adding an `.on('change')` event to the `depends on` property. Whenever a value is selected we will iterate over all other `depends on` properties that have a possible option targeting the recently changed one and mark it as disabled. 

This way, the user will not be able to create a dependency with a property that already has a value selected in the `depends on` property.

### CODE REVIEW

**Review by habemuscode — APPROVED** [review]
LGTM

File: views/js/form/depends-on-property.js [hunk 1]
```diff
@@ -71,9 +71,14 @@
         ];
     }
 
+    function hasAnyDependency($container) {
```
Comment (adriloma) [inline views/js/form/depends-on-property.js:74]: missing jsdoc

File: views/js/uiForm.js [hunk 2]
```diff
@@ -775,21 +776,37 @@
...
+                            dependsOn.toggle($dependsOnSelect, $dependsOnSelect.parent(), $this.closest('.property-edit-container'));
                         } else {
-                            dependsOnSelect.parent().hide();
+                            $dependsOnSelect.parent().hide();
                         }
                     }
                 });
             }
 
+            function filterDependsOnProperty() {
```
Comment (adriloma) [inline views/js/uiForm.js:797]: misssed jsdoc.

**Review by adriloma — APPROVED** [review]
Code review only, I suggest document every new function.

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following conventional commits
- [x] Commits messages are meaningful
- [x] Commits are atomic

**Comment by AntonNeviarouski** [comment]
QA verified

**Pull request merged**